src/components/flag/code/views/flagView.jsx

```diff
@@ -0,0 +1,14 @@
+var React = require('react');
+
+module.exports = React.createClass({
+
+  propTypes: {
+    purpose: React.PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'info']),
+    size: React.PropTypes.oneOf(['small', 'large', 'extra-large']),
+    position: React.PropTypes.oneOf(['top', 'right', 'bottom', 'left'])
+  },
+
+  render: function() {
+    return require('../templates/flagTemplate.jsx')(this.props);
```

Comment:
A note for future reference – I think this pattern is going to very quickly need to be revisited. `this.state` is not passed here and it's most definitely going to be needed by sub-rendering functions at some point.